sql/core/src/main/scala/org/apache/spark/sql/execution/command/views.scala

```diff
@@ -268,30 +268,32 @@ case class AlterViewAsCommand(
     name: TableIdentifier,
     originalText: String,
     query: LogicalPlan) extends RunnableCommand {
+  require(query.resolved)
```

Comment:
This is not needed. `query` is a child now, and the analyzer will make sure the entire query plan tree is analyzed.